dspy/teleprompt/gepa/gepa_utils.py

```diff
@@ -100,9 +100,15 @@ def custom_propose_new_texts(
 
             self.propose_new_texts = custom_propose_new_texts
 
+        #Logger
+        self.logger = logging.getLogger(__name__)
```

Comment:
Have a look at how the logger is used elsewhere. Better to declare it at module level as done in other dspy files.